### PR TITLE
Add security scanning Action

### PR DESCRIPTION
**Pull request by davidrans**
Let's see if this is useful

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
```
Comment (ardelato) [inline .github/workflows/codeql-analysis.yml:34]: Do we need this matrix if we are only checking for one language?

**Review by ardelato — APPROVED** [review]
CR 🦖

**Comment by k0rvusk0r4x** [comment]
Ideas on the best way to test this?

**Comment by ardelato** [comment]
> Ideas on the best way to test this?

I think just validating that the GitHub Actions passed would be good?

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Looks like this step failed
CleanShot 2022-08-08 at 12 30 33

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (davidrans) [reply .github/workflows/codeql-analysis.yml:34]: I figured I'd just use the suggested default unless it didn't work for some reason

**Comment by davidrans** [comment]
I guess re-running fixed it

**Pull request merged**